config/api/src/main/java/org/jboss/windup/config/operation/Iteration.java

```diff
@@ -208,6 +242,29 @@ public void perform(GraphRewrite event, EvaluationContext context)
     {
         Variables variables = Variables.instance(event);
         Iterable<? extends WindupVertexFrame> frames = getSelectionManager().getFrames(event, context);
+
+        boolean hasCommitOperation = hasCommitOperation(operationPerform) || hasCommitOperation(operationOtherwise);
+        boolean hasIterationOperation = hasIterationProgress(operationPerform) || hasIterationProgress(operationOtherwise);
+        DefaultOperationBuilder commit = new NoOp();
+        DefaultOperationBuilder iterationProgressOperation = new NoOp();
+
+        if (!hasCommitOperation || !hasIterationOperation)
+        {
+            int frameCount = Iterables.size(frames);
+            if (frameCount > 100)
+            {
+                if (!hasCommitOperation)
+                    commit = Commit.every(100);
```

Comment:
I didn't directly test the performance, but in general having large amounts of uncommitted data will slow Tinkerpop down.